src/toolkit/position.cycpp.h

```diff
@@ -0,0 +1,32 @@
+
+// This includes the required header to add geographic coordinates to a
+// archetypes.
+// One only need to:
+// - '#include "toolkit/position.cycpp"' in the core of the archetype class (as
+// private)
+// - and in the EnterNotify() method:
+//   - set the coordinates 'coordinates = cyclus::toolkit::Position(latitude,
+//   longitude);'
+//   - call the record method: 'coordinates.RecordPosititon(this);'
+
+cyclus::toolkit::Position coordinates;
+
+#pragma cyclus var { \
+"default": 0.0, \
+"uilabel": "Geographical latitude in degrees as a double", \
+"doc": "Latitude of the agent's geographical position. The value should " \
+       "be expressed in degrees as a double." \
+}
+double latitude;
+// required for compilation but not added by the cycpp preprocessor...
+std::vector<int> cycpp_shape_latitude;
```

Comment:
What is this for?

Reply:
this is because `cycpp` is not run on the file generate by the `cpp`

see https://github.com/cyclus/cyclus/blob/279a758fbfb33f0bf142e0126bf269673c3364e7/cmake/UseCyclus.cmake#L122
and 
https://github.com/cyclus/cyclus/blob/279a758fbfb33f0bf142e0126bf269673c3364e7/cli/cycpp.py#L2328-L2335

so when using this snippet method we miss generated code for the additional included variables...
I try to change the `--pass3-use-orig` to `--pass3-use-pp` and `cyclus` is not compiling :(

I don't know what the best solution is...

Reply:
cc @gonuke this is a follow-up on the conversation we had this morning...